src/m365/commands/setup.ts

```diff
@@ -106,59 +205,218 @@ class SetupCommand extends AnonymousCommand {
+    if (!preferences.summary) {
+      return;
+    }
 
+    // used only for testing. Normally, we'd get the settings from the answers
+    /* c8 ignore next 3 */
+    if (!settings) {
+      settings = this.getSettings(preferences);
+    }
+
+    await logger.logToStderr('');
+    await logger.logToStderr('Configuring settings...');
+    await logger.logToStderr('');
+
+    await this.configureSettings({ preferences, settings, silent: false, logger });
+
+    if (!this.verbose) {
       await logger.logToStderr('');
-      await logger.logToStderr('Configuring settings...');
-      await logger.logToStderr('');
+      await logger.logToStderr(chalk.green('DONE'));
+    }
+  }
 
-      await this.configureSettings(settings, false, logger);
+  private async configureExistingEntraApp(logger: Logger): Promise<Preferences> {
+    await logger.logToStderr('Please provide the details of the existing app registration.');
+    let clientCertificateFile: string | undefined;
+    let clientCertificateBase64Encoded: string | undefined;
+    let clientCertificatePassword: string | undefined;
 
-      if (!this.verbose) {
-        await logger.logToStderr('');
-        await logger.logToStderr(chalk.green('DONE'));
+    const clientId = await cli.promptForInput({
+      message: 'Client ID:',
+      /* c8 ignore next */
+      validate: value => value.length > 0
+    });
+    const tenantId = await cli.promptForInput({
+      message: 'Tenant ID (leave common if the app is multitenant):',
+      default: 'common'
+    });
+    const clientSecret = await cli.promptForInput({
+      message: 'Client secret (leave empty if you use a certificate or a public client):'
+    });
+    if (!clientSecret) {
+      clientCertificateFile = await cli.promptForInput({
+        message: `Path to the client certificate file (leave empty if you want to specify a base64-encoded certificate string):`
+      });
+      if (!clientCertificateFile) {
+        clientCertificateBase64Encoded = await cli.promptForInput({
+          message: 'Base64-encoded certificate string:'
+        });
       }
+      if (clientCertificateFile || clientCertificateBase64Encoded) {
+        clientCertificatePassword = await cli.promptForInput({
+          message: 'Password for the client certificate (leave empty if the certificate is not password-protected):'
+        });
+      }
+    }
+
+    return {
+      clientId,
+      tenantId,
+      clientSecret,
+      clientCertificateFile,
+      clientCertificateBase64Encoded,
+      clientCertificatePassword
+    };
+  }
+
+  private async createNewEntraApp(preferences: Preferences, logger: Logger): Promise<AppInfo> {
+    if (!await cli.promptForConfirmation({
+      message: 'CLI for Microsoft 365 will now sign in to your Microsoft 365 tenant as Microsoft Azure CLI to create a new app registration. Continue?',
+      default: false
+    })) {
+      throw new Error();
     }
+
+    // logout, just in case
+    await auth.clearConnectionInfo();
```

Comment:
Hi @waldekmastykarz, this line here deletes all connections that I've already created. That kinda defeats the purpose of the multi-account setup. I may want to create multiple apps in multiple tenants. I don't want to loose my existing details, I want to be able to switch back afterwards.

Reply:
I added it to ensure no conflicts with existing connections, but I see your point. I'll revert this change.